pendulum_controller/test/test_pendulum_controller_node.cpp

```diff
@@ -0,0 +1,109 @@
+  rclcpp::NodeOptions node_options;
+  std::vector<double> feedback_matrix = {-10.0000, -51.5393, 356.8637, 154.4146};
+
+  static void SetUpTestCase()
+  {
+    ASSERT_FALSE(rclcpp::ok());
+    rclcpp::init(0, nullptr);
+    ASSERT_TRUE(rclcpp::ok());
+  }
+
+  void SetUp() override
+  {
+    params.emplace_back("state_topic_name", "joint_states");
+    params.emplace_back("command_topic_name", "command");
+    params.emplace_back("teleop_topic_name", "setpoint");
+    params.emplace_back("command_publish_period_us", 10000);
+    params.emplace_back("enable_topic_stats", false);
+    params.emplace_back("topic_stats_topic_name", "controller_stats");
+    params.emplace_back("topic_stats_publish_period_ms", 1000);
+    params.emplace_back("deadline_duration_ms", 0);
+    params.emplace_back("controller.feedback_matrix", feedback_matrix);
+  }
+
+  static void TearDownTestCase()
+  {
+    (void)rclcpp::shutdown();
+  }
+};
+
+TEST_F(InitNodesTest, test_options_constructor) {
+  node_options.parameter_overrides(params);
+
+  const auto test_node =
+    std::make_shared<PendulumControllerNode>(node_options);
+
+  // auto names = test_node->get_node_names();
+
+  EXPECT_STREQ("pendulum_controller", test_node->get_name());
+  // EXPECT_STREQ("/ns", test_node->get_namespace());
+  // EXPECT_STREQ("/ns/my_node", test_node->get_fully_qualified_name());
+  // EXPECT_EQ(names.size(), 1u);
+  // EXPECT_STREQ(names[0].c_str(), "/pendulum_controller");
+  // EXPECT_STREQ("/", test_node->get_namespace());
+}
+
+TEST_F(InitNodesTest, test_transition) {
+  node_options.parameter_overrides(params);
+
+  const auto test_node =
+    std::make_shared<PendulumControllerNode>(node_options);
+
+  EXPECT_EQ(State::PRIMARY_STATE_UNCONFIGURED, test_node->get_current_state().id());
+  ASSERT_EQ(
+    State::PRIMARY_STATE_INACTIVE, test_node->trigger_transition(
+      rclcpp_lifecycle::Transition(Transition::TRANSITION_CONFIGURE)).id());
+  ASSERT_EQ(
+    State::PRIMARY_STATE_ACTIVE, test_node->trigger_transition(
+      rclcpp_lifecycle::Transition(Transition::TRANSITION_ACTIVATE)).id());
+  ASSERT_EQ(
+    State::PRIMARY_STATE_INACTIVE, test_node->trigger_transition(
+      rclcpp_lifecycle::Transition(Transition::TRANSITION_DEACTIVATE)).id());
+  ASSERT_EQ(
+    State::PRIMARY_STATE_UNCONFIGURED, test_node->trigger_transition(
+      rclcpp_lifecycle::Transition(Transition::TRANSITION_CLEANUP)).id());
+  ASSERT_EQ(
+    State::PRIMARY_STATE_FINALIZED, test_node->trigger_transition(
+      rclcpp_lifecycle::Transition(Transition::TRANSITION_UNCONFIGURED_SHUTDOWN)).id());
+}
+
+TEST_F(InitNodesTest, test_param_constructor) {
+  node_options.parameter_overrides(params);
+
+  const auto test_node_ptr =
+    std::make_shared<PendulumControllerNode>(node_options);
+  auto names = test_node_ptr->get_node_names();
+
+  // EXPECT_EQ(names.size(), 1u);
+  // EXPECT_STREQ(names[0].c_str(), "/pendulum_controller");
+  // EXPECT_STREQ("/", test_node_ptr->get_namespace());
+}
```

Comment:
Add a test to check all the required params exist. e.g. https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/blob/0d12a28fe7099a84bacf953c6b6b6093b372aba9/src/perception/segmentation/euclidean_cluster_nodes/test/test_euclidean_cluster_nodes.hpp#L41-106